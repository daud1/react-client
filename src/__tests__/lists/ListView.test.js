import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ListView from '../../components/lists/listView.component';
import shallowToJson from 'enzyme-to-json';
import '../../setupTests';

describe ('ListView Component', () => {

    it('renders without crashing', () => {
        localStorage.setItem("token", "random_string");
        const wrapper = shallow(<ListView />);
        expect(wrapper.exists(<div></div>)).toBe(true)
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<ListView />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
})