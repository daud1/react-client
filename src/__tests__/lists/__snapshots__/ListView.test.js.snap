// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView Component matches snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-default"
  >
    <div
      class="container-fluid"
    >
      <div
        class="navbar-header"
      >
        <Link
          className="navbar-brand"
          replace={false}
          to="/lists"
        >
          Shopping Lists
        </Link>
      </div>
      <div
        class="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1"
      >
        <ul
          class="nav navbar-nav"
        >
          <Search
            callback={[Function]}
            setValue={[Function]}
          />
        </ul>
        <ul
          class="nav navbar-nav navbar-right"
        >
          <br />
          <withRouter(LogOut) />
        </ul>
      </div>
    </div>
  </nav>
  <CreateList
    callback={[Function]}
  />
  <h3>
     See all your lists here: 
  </h3>
  <Lister
    callback={[Function]}
  />
  <nav>
    <ul
      className="pagination"
    />
  </nav>
</div>
`;
